lab-dan/server.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+
+const PORT = process.env.PORT || 3000;
+
+function start (chatClient, cb) {
+  let pool = [];
+  const ee = new EE();
+
+  let server = net.createServer();
+
+  server.on('connection', function (socket) {
+    let client = chatClient(socket);
+    pool.push(client);
+    socket.on('data', function (data) {
+      console.log('data received from client!');
+      console.log('data:', data.toString());
+      const command = data.toString().split(' ').shift().trim();
+      console.log(command.startsWith('\\'));
+      if (command.startsWith('\\')) {
+        console.log('emitting event');
+        let message = data.toString().split(' ').slice(1).join(' ');
+        console.log('command:', command, '\nmessage:', message);
+        ee.emit(command, client, message, pool);
+      } else {
+        ee.emit('default', client);
+      }
+    });
+  });
+
+  server.listen(PORT, function() {
+    console.log('server started on', PORT);
+    if(cb) {
+      cb(server);
+    }
+  });
+
+  ee.on('default', function (client) {
+    console.log('default event');
+    client.socket.write('not a command');
+  });
+
+  ee.on('\\\\all', function (client, message, clientPool) {
```

Comment:
So your wack commands are supposed to have two backslashes, correct? Your preference?